python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -172,124 +178,46 @@ def get_in_progress_runs_by_step(graphene_info, job_names, step_keys):
     ]
 
 
-def get_asset_run_stats_by_step(graphene_info, job_names, asset_nodes):
-    # This is a utility method that gets the latest run that selected an asset,
-    # by searching within the last 5 runs of each job the asset belongs in.
-    # If none of the most recent runs selected the asset, we return back a GrapheneJobRunsCount
-    # object that contains the total number of job runs that have occurred since the latest
-    # asset materialization (or the total number of job runs if the asset has never been
-    # materialized).
+def get_asset_run_stats_by_step(graphene_info, asset_nodes):
+    latest_run_by_step = get_latest_asset_run_by_step_key(graphene_info, asset_nodes)
 
-    latest_run_by_step = get_latest_asset_run_by_step_key(graphene_info, job_names, asset_nodes)
-    assets_to_fetch_run_count = [
-        asset_node
-        for asset_node in asset_nodes
-        if latest_run_by_step.get(asset_node.op_name) == None
-    ]
-    jobs_runs_count = get_asset_runs_count_by_step(graphene_info, assets_to_fetch_run_count)
-
-    return [
-        latest_run_by_step.get(asset_node.op_name, jobs_runs_count.get(asset_node.op_name))
-        for asset_node in asset_nodes
-    ]
+    return [latest_run_by_step.get(asset_node.op_name) for asset_node in asset_nodes]
 
 
-def get_latest_asset_run_by_step_key(graphene_info, job_names, asset_nodes):
+def get_latest_asset_run_by_step_key(graphene_info, asset_nodes):
     from ..schema.pipelines.pipeline import GrapheneLatestRun, GrapheneRun
 
-    # This method returns the latest run that has occurred for a given step.
-    # Because it is expensive to deserialize PipelineRun objects, we limit this
-    # query to retrieving the last 5 runs per job. If no runs have occurred, we return
-    # a GrapheneLatestRun object with no run. If none of the latest runs contain the
-    # step key, we return None.
-
     instance = graphene_info.context.instance
 
     latest_run_by_step: Dict[str, PipelineRun] = {}
 
-    run_records_by_job = {
-        job_name: instance.get_run_records(RunsFilter(pipeline_name=job_name), limit=5)
-        for job_name in job_names
-    }
-
-    get_step_keys = lambda record: graphene_info.context.instance.get_execution_plan_snapshot(
-        record.pipeline_run.execution_plan_snapshot_id
-    ).step_keys_to_execute
-
-    # step_keys list is in same order as records list in get_step_keys
-    record_step_keys_by_job = {
-        job_name: [get_step_keys(record) for record in run_records]
-        for job_name, run_records in run_records_by_job.items()
-    }
-
-    for asset_node in asset_nodes:
-        asset_job_names = asset_node.job_names
-        step_key = asset_node.op_name
-
-        total_num_records = sum(
-            [len(run_records_by_job.get(job_name, [])) for job_name in asset_job_names]
+    latest_run_id_by_asset: Dict[AssetKey, str] = {}
+    for asset in asset_nodes:
+        event_records = instance.get_event_records(
```

Comment:
Should highlight that we're now doing roundtrips to the DB on the order of N, where N is the number of nodes in the graph.

How does this feel on mega-graphs?  Maybe we can start updating the asset key table on writes instead of querying the event log directly?